### ISSUE TITLE
System.Exception loses Message value during serialization/deserialization

### ISSUE THREAD

**Issue by eric-davis** [dotnet/corefx#20776]
System.Exception appears to have some serialization issues. If I perform the following simple test utilizing Newtonsoft.Json for serialization, it fails:

```c#
            var ex = new Exception("Something bad happened!!!");
            var json = JsonConvert.SerializeObject(ex);
            var obj = JsonConvert.DeserializeObject<Exception>(json);
            Assert.AreEqual(ex.Message, obj.Message);

```

The custom message value is lost during deserialization and replaced with the default exception message: "Exception of type 'System.Exception' was thrown."

This was tested against System.Runtime version 4.1.0.0

**Comment by stephentoub** [comment]
Seems like this is an issue for https://github.com/JamesNK/Newtonsoft.Json?

**Comment by karelz** [comment]
@eric-davis can you please move your bug report there?

**Comment by tarekgh** [comment]
I have reported the issue to https://github.com/JamesNK/Newtonsoft.Json/issues/1336 and closing the issue here.

**Comment by JamesNK** [comment]
In full .NET framework Exceptions are deserialized using the ISerializable constructor. That isn't available in .NET Core.

The only way I can think of deserializing information is forcing the serializer to use `ctor(string, Exception)` to create the object on types that inherit from Exception. Is there a better way?

**Comment by JamesNK** [comment]
Another option might be to set the private `_message` and `_innerException` fields. Would there be issues with this on .NET Core? Although if there is no ISerializable in .NET Core with `Exception` then those fields might be `readonly`... might that cause problems?

**Comment by JamesNK** [comment]
Could we temporarily reopen this issue while we discuss approaches to deserializing Exceptions with .NET Core?

**Comment by karelz** [comment]
Agreed. Let's reopen to flush out the plan fully.

**Comment by svick** [comment]
It seems to me that this is already fixed in 2.0:

```
> dotnet run -f netcoreapp1.1
Exception of type 'System.Exception' was thrown.
> dotnet run -f netcoreapp2.0
Something bad happened!!!
```

**Comment by tarekgh** [comment]
@svick is right, the ISerializable is added there 

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Exception.cs#L32

@eric-davis let's know if there is anything not working with v2.0. Thanks.
